aptos-move/aptos-vm/src/transaction_validation.rs

```diff
@@ -101,7 +104,10 @@ pub(crate) fn run_script_prologue(
     traversal_context: &mut TraversalContext,
     is_simulation: bool,
 ) -> Result<(), VMStatus> {
-    let txn_sequence_number = txn_data.sequence_number();
+    let txn_sequence_number = txn_data
+        .replay_protector()
+        .get_sequence_number()
+        .unwrap_or(0);
```

Comment:
this should be an error (that results in txn discard) instead of unwrap? otherwise a Nonce txn could be executed as a sequence number 0?